colossus-tests/src/test/scala/colossus/controller/PipeCombinatorSpec.scala

```diff
@@ -0,0 +1,82 @@
+package colossus.controller
+
+import org.scalatest.{TryValues, OptionValues, MustMatchers, WordSpec}
+
+import scala.util.{Failure, Success}
+
+class PipeCombinatorSpec extends WordSpec with MustMatchers with OptionValues with TryValues{
+
+  "PipeCombinator" must {
+
+    "combine 2 pipes with a map function" in {
+
+      import ReflexiveCopiers._
+
+      val writeSink: Pipe[String, String] = new InfinitePipe[String](100)
+      val readSource: Pipe[Int, Int] = new InfinitePipe[Int](100)
+
+      val map: (String) => Seq[Int] = (s: String) => s.split(":").map(_.toInt).toSeq
+
+      val combined: Pipe[String, Int] = writeSink.join(readSource)(map)
+      combined.push("1:2:3")
+      combined.push("4:5")
+      combined.complete()
+
+      combined.fold(Seq[Int]())((el, ac) => ac :+ el).execute {
+        case Success(x) => x must equal(Seq(1, 2, 3, 4, 5))
```

Comment:
It's possible this test is still not actually working, you should create a variable outside the callback and set its value inside the execute closure, then check the value after calling execute.  Right now if the execute closure throws an exception, it may actually be getting suppressed (like what would happen if you did it inside a future)